craft_application/models/project.py

```diff
@@ -131,13 +130,14 @@ def _vectorise_architectures(cls, values: str | list[str]) -> list[str]:
     @pydantic.field_validator("build_on", "build_for", mode="after")
     @classmethod
     def _validate_architectures(cls, values: list[str]) -> list[str]:
-        """Validate the architecture entries."""
-        for architecture in values:
-            if architecture != "all" and not is_valid_architecture(architecture):
-                raise errors.CraftValidationError(
-                    f"Invalid architecture: {architecture!r} "
-                    "must be a valid debian architecture."
-                )
+        """Validate the architecture entries.
+
+        Entries must be a valid debian architecture or 'all'. Architectures
+        may have an optional base prefix formatted as '[<base>:]<arch>'.
+
+        :raises ValueError: If any of the architectures are not valid.
+        """
+        [craft_platforms.parse_base_and_architecture(arch) for arch in values]
```

Comment:
Craft-application is in a weird spot because the model allows for `powerpc` but its build planner from craft-platforms does not. 

So, I think we can consider this a safe change. It has 2 effects:

1. Raise a ValueError instead of a CraftValidationError, which means better error presentation via pydantic
2. For `powerpc`, an error is raised on validation rather than when creating the build plan.